test_runner/src/main.rs

```diff
@@ -1,3 +1,155 @@
+use core::time;
+use std::thread;
+
+use althea_kernel_interface::{KernelInterfaceError, KI};
+
 fn main() {
-    println!("Hello, world!");
+    /*
+    These are currently connected as such:
+    A---------B
+     \       /|
+      \     / |
+       \   /  |
+        \ /   |
+         X    |
+        / \   |
+       /   \  |
+      /     \ |
+     /       \|
+    D---------C
+    */
+    let testa = ("nA".to_string(), 0);
+    let testb = ("nB".to_string(), 1);
+    let testc = ("nC".to_string(), 2);
+    let testd = ("nD".to_string(), 3);
+
+    let ten_mins = time::Duration::from_secs(600);
+
+    let namespaces = NamespaceInfo {
+        names: vec![
+            testa.clone().0,
+            testb.clone().0,
+            testc.clone().0,
+            testd.clone().0,
+        ],
+        linked: vec![
+            // arbitrary connections 
+            (testa.clone(), testb.clone()),
+            (testb.clone(), testc.clone()),
+            (testa, testc.clone()),
+            (testc, testd.clone()),
+            (testb, testd),
+        ],
+    };
+    let res = setup_ns(namespaces);
+    println!("Namespaces setup: {:?}", res);
+    // this sleep is for debugging so that the container can be accessed to poke around in
+    thread::sleep(ten_mins);
```

Comment:
we probably want to remove this for CI, otherwise we're just burning CI minutes.